File: languages/go/goodbot/data_loader.go
package goodbot

import (
	"encoding/json"
	"fmt"
	"os"
)

// BotData represents the structure of our JSON data. Adjust this according to your actual JSON structure.
type BotData map[string]map[string]struct {
	UserAgent        string   `json:"UserAgent"`
	Method           string   `json:"Method"`
	ValidDomains     []string `json:"ValidDomains"`
	BypassFlag       string   `json:"BypassFlag,omitempty"`
	UserAgentPattern string   `json:"UserAgentPattern,omitempty"`
}

// LoadBotData loads bot data from a specified JSON file path.
func LoadBotData(filePath string) (BotData, error) {
	var botData BotData
	file, err := os.Open(filePath)
	if err != nil {
		return nil, fmt.Errorf("error opening bot data file: %v", err)
	}
	defer file.Close()

	decoder := json.NewDecoder(file)
	if err := decoder.Decode(&botData); err != nil {
		return nil, fmt.Errorf("error decoding bot data: %v", err)
	}

	return botData, nil
}

File: languages/python/goodbot/data_loader.py
import json
from pathlib import Path

def load_bot_data():
    local_data_path = Path(__file__).parent.parent.parent.parent / "data/goodbots.json"
    pkg_data_path = Path(__file__).parent / "data/goodbots.json"
    if local_data_path.exists():
        data_path = local_data_path
    else:
        data_path = pkg_data_path
    with open(data_path, 'r') as file:
        data = json.load(file)
    return data


data = load_bot_data()
print(data)